commit d985a24acc4c5dae68c1301a4368f0fac08bb47c
Author: Jean Honlet <jehon@users.noreply.github.com>
Date:   Fri Mar 8 13:59:00 2019 +0100

    Fix #1651: add support for nodejs experimental-modules stacktraces

diff --git a/lib/jasmine-core/jasmine.js b/lib/jasmine-core/jasmine.js
index 60d29de7..68ecef56 100644
--- a/lib/jasmine-core/jasmine.js
+++ b/lib/jasmine-core/jasmine.js
@@ -6463,9 +6463,21 @@ getJasmineRequireObj().SpyStrategy = function(j$) {
 
 getJasmineRequireObj().StackTrace = function(j$) {
   function StackTrace(error) {
-    var lines = error.stack
-      .split('\n')
-      .filter(function(line) { return line !== ''; });
+    var lines = [ 'unknown result!' ];
+    if (Array.isArray(error.stack)) {
+      /**
+       * node --experimental-modules (v11.9 at least)
+       * esm stackstraces are are composed of an array of "CallSite"
+       *
+       * TODO: See later if this behavior need to be adapted again
+       */
+      lines = error.stack.map(function(site) { return site.toString(); });
+    } else {
+      /* normal handling */
+      lines = error.stack
+        .split('\n')
+        .filter(function(line) { return line !== ''; });
+    }
 
     var extractResult = extractMessage(error.message, lines);
 
